crypto/s2n_sequence.c

```diff
@@ -29,10 +29,12 @@ int s2n_increment_sequence_number(struct s2n_blob *sequence_number)
         if (sequence_number->data[i]) {
             break;
         }
-
-        /* RFC 5246 6.1: If a TLS implementation would need to wrap a sequence number, it must
-         * renegotiate instead. We don't support renegotiation. Caller needs to create a new session.
-         * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
```

Comment:
To be clear, we still never trigger renegotiation, and certainly not because the record number cap was reached. It's an unstable feature only available for clients and currently purely used for delayed client auth.

Reply:
I updated the doc with more clarification. For learning purpose, why would some clients want to delay their auth?

Reply:
The theoretical main use case for renegotiation is: A client and a server handshake, without client auth. Then the client requests some data that requires client auth. To authenticate the client, the server initiates renegotiation, and the second handshake does client auth. So it's not really that the clients want to delay their auth, it's that the servers don't initially request the auth.

Reply:
Interesting. Is the reason servers don't initially request authentication to save computational resources?

Reply:
I think it's theoretically because not all clients request resources that require client auth, so not all clients need to authenticate.